src/RA_Dlg_AchEditor.cpp

```diff
@@ -714,481 +712,362 @@ INT_PTR Dlg_AchievementEditor::AchievementEditorProc(HWND hDlg, UINT uMsg, WPARA
 {
     BOOL bHandled = TRUE;
 
-    switch (uMsg)
+    // Message handlers here are only to prevent necessary casting
+    // TRANSITION: These message handlers can be moved to use the ViewModel logic as long as casting is minimized
+    const auto OnGetMinMaxInfo = []([[maybe_unused]] HWND, MINMAXINFO* restrict lpMinMaxInfo) noexcept
     {
-        case WM_TIMER:
-        {
-            // Ignore if we are currently editing a box!
-            if (g_hIPEEdit != nullptr)
-                break;
+        lpMinMaxInfo->ptMinTrackSize = pDlgAchEditorMin;
+    };
 
-            Achievement* pActiveAch = ActiveAchievement();
-            if (pActiveAch == nullptr)
+    // hwndCtl depends on id
+    const auto OnCommand = [this, &bHandled](HWND hwnd, int id, HWND hwndCtl, UINT codeNotify)
```

Comment:
I was able to use all the values supplied by the Windows API here and a few other places.